File: android-app/src/main/java/org/example/app/units/GifsListUnitsFactory.kt
/*
 * Copyright 2019 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
 */

package org.example.app.units

import dev.icerock.moko.units.UnitItem
import org.example.app.TileNews
import ru.alex009.library.feature.gifs.presentation.GifsListViewModel

class GifsListUnitsFactory : GifsListViewModel.UnitsFactory {
    override fun createGifTile(id: Long, gifUrl: String): UnitItem {
        return TileNews().apply {
            itemId = id
            this.title = gifUrl
        }
    }
}

File: ios-app/src/units/GifsListUnitsFactory.swift
/*
 * Copyright 2019 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
 */

import UIKit
import MultiPlatformLibrary
import MultiPlatformLibraryUnits

class GifsListUnitsFactory: GifsListViewModelUnitsFactory {
    func createGifTile(id: Int64, gifUrl: String) -> UnitItem {
        return UITableViewCellUnit<NewsTableViewCell>(
            data: NewsTableViewCell.CellModel(
                id: id,
                title: gifUrl,
                description: gifUrl
            ),
            configurator: nil
        )
    }
}
